src/hardware/voodoo.cpp

```diff
@@ -6996,6 +7046,37 @@ static constexpr uint32_t voodoo_r(const uint32_t addr)
 	return 0xffffffff;
 }
 
+int get_num_processors()
+{
+	// The C++ standard provides an honest return value of 0 if it can't figure
+	// it out, so we prefer that first and can inform the user with warning.
+
+	if (auto count = std::thread::hardware_concurrency(); count > 0) {
+		return static_cast<int>(count);
+	}
+	LOG_WARNING("VOODOO: Failed getting the number of host processors");
+
+	// SDL uses similar API calls with some variations, however it will lie
+	// and return 1 even if it fails. So we only use it as a fallback and
+	// accept whatever it gives us.
+
+	return SDL_GetCPUCount();
```

Comment:
Better, but why not just use `SDL_GetCPUCount()` and be done with it? 😆 The C/C++ standard is sometimes a bit silly because it needs to support all sorts of weirdo hardware... I'd bet large sums of money `SDL_GetCPUCount()` is spot on all the time on the much narrower platforms supported by SDL.

Reply:
Yeah.. I've added a log of the number of threads detected, so combined with the manual setting, worst case for a mis-detect is the user had to punch in their desired number of threads (and that's going to pretty rare!)